### ISSUE TITLE
gherkin (Ruby): Undocumented API changes

### ISSUE THREAD

**Issue by enkessler** [cucumber/common#1559]
## Summary

I'm in the process of updating one of my projects to be able to use the 19.x major version of the `cucumber-gherkin` gem and am running into some unexpected API changes that do not appear to be reflected in the [CHANGELOG](https://github.com/cucumber/common/blob/master/gherkin/CHANGELOG.md) at all.

## Expected Behavior

Because the CHANGELOG only mentions some dialect updates (which would not impact my project) and dependency upgrades (whose own changelogs, in turn, had no impacting call outs), I was expecting things to still 'just work' after I upgraded from 18.x to 19.x.

## Current Behavior

I'm seeing several places in the code where data keys in returned AST data got converted from `snake_case` to `camelCase` which, naturally, breaks any code that uses that particular portion of the API. So far, I've run into

`parse_error` -> `parseError`
`media_type` -> `mediaType`
`doc_string` -> `docString`
`data_table` -> `dataTable`

but I'm still in the process of upgrading so there may be several more that I'll run into.

## Possible Solution

It depends on if all of the name changes were intentional or not. It's not the first time that the names of things have changed but I don't know if it was an attempt at standardization across language implementations or it's just that whoever wrote the code was in the habit of writing in a different style.

If the changes were intentional
 - Update the CHANGELOG to reflect the changes

If the changes were not intentional
 - Revert them and probably add some more testing around what the AST structure is so that these kinds of bugs don't slip through in the future

## Context & Motivation

While I understand that an increase in the major version number indicates backwards incompatibility in some fashion, those incompatibilities tend to get called out. Since they weren't mentioned, I figured that I'd check and see if the changes were just as much of a surprise to everyone else as they were to me.

## Your Environment

* Version used: `cucumber-gherkin (19.0.2)`
* Operating System and version: N/A
* Link to your project: https://github.com/enkessler/cuke_modeler

**Comment by aurelien-reeves** [comment]
Hello @enkessler 

Thanks for the report.

This is actually related to messages@16.0. It does not use protocol buffers anymore, but jsonschema to describe the messages.

The impact on AST is more a side effect than an expected change.

We need to improve the documentation, and the changelog, accordingly.

Regarding Ruby, the big difference is that messages are no longer classes, but hashes, and the key are now camelCase as you described.

**Comment by aurelien-reeves** [comment]
CHANGELOGs have been updated
Thanks again for your report

**Comment by enkessler** [comment]
> This is actually related to messages@16.0

Yeah, I thought that might be the case but I didn't see anything in its CHANGELOG when I looked and when I searched in this repository for the new names, I found them under the `gherkin` sub-project, so I figured that this is where they came from (even if those changes were in response to something else further down the stack). In any case, I've now got my answer: It was intentional.

Thanks for the info and for updating the relevant CHANGELOG!
